registry/remote/registry_example_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package registry_test gives examples code of functions in the remote package.
+package remote_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+var exampleRegistry *remote.Registry
+
+func TestMain(m *testing.M) {
+	// Mocking local registry
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+	defer ts.Close()
+	exampleUri, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+	exampleRegistry, err = remote.NewRegistry(exampleUri.Host) // Create a registry via the remote host
+	if err != nil {
+		panic(err) // Handle error
+	}
+	exampleRegistry.PlainHTTP = true // Use HTTP
+	os.Exit(m.Run())
+}
+
+// ExampleRegistry_Repositories gives example snippets for listing respositories in the registry with pagination.
+func ExampleRegistry_Repositories() {
+	fn := func(repos []string) error { // Setup a callback function to process returned repository list
+		for _, repo := range repos {
+			fmt.Println(repo)
+		}
+		return nil
+	}
+
+	ctx := context.Background()
+	err := exampleRegistry.Repositories(ctx, fn)
```

Comment:
rename to reg or something along the lines of what the go-lang styles recommend. 
Also wonder if we should bring in the registry creation `remote.NewRegistry(exampleUri.Host) ` into this part.